ts/components/SectionStatusComponent.tsx

```diff
@@ -64,19 +74,35 @@ const SectionStatusComponent: React.FC<Props> = (props: Props) => {
     return null;
   }
 
+  const viewRef = React.createRef<View>();
   const sectionStatus = props.sectionStatus;
   const iconName = statusIconMap[sectionStatus.level];
   const backgroundColor = statusColorMap[sectionStatus.level];
   const locale = getFullLocale();
   const maybeWebUrl = maybeNotNullyString(
     sectionStatus.web_url && sectionStatus.web_url[locale]
   );
+
+  React.useEffect(() => {
+    if (viewRef.current) {
+      props.onSectionRef?.(viewRef);
+    }
+
+    const unsubscribe = props.navigationProps?.addListener("didFocus", () => {
+      if (viewRef.current) {
+        props.onSectionRef?.(viewRef);
+      }
+    });
+
+    return () => unsubscribe?.remove();
+  }, [viewRef]);
+
```

Comment:
```suggestion
  const handleOnSectionRef = () => {
    if (viewRef.current) {
      props.onSectionRef?.(viewRef);
    }
  };

  React.useEffect(() => {
    handleOnSectionRef();
    const unsubscribe = props.navigationProps?.addListener(
      "didFocus",
      handleOnSectionRef
    );
    return () => unsubscribe?.remove();
  }, [viewRef]);
```
Could it save some code repetition?

Reply:
Super! PR allineata come consigliata ✅